python/vyos/util.py

```diff
@@ -903,14 +903,16 @@ def get_bridge_fdb(interface):
     tmp = loads(cmd(f'bridge -j fdb show dev {interface}'))
     return tmp
 
-def get_interface_config(interface):
+def get_interface_config(interface, netns=None):
     """ Returns the used encapsulation protocol for given interface.
         If interface does not exist, None is returned.
     """
-    if not os.path.exists(f'/sys/class/net/{interface}'):
+    from vyos.util import run
+    netns_exec = f'ip netns exec {netns}' if netns else ''
+    if run(f'{netns_exec} test -e /sys/class/net/{interface}') != 0:
```

Comment:
I don't know how much vyos prefers executing commands vs direct syscalls, but when I looked at `pyroute2` recently, it showed me that it was pretty simple to implement a context manager in which executing python code would be executed within another network namespace- no need for any `ip netns exec` or forking of any kind

Reply:
I remember @c-po against `pyroute2`, and we use our modules.